src/components/ftb-stage-table/ftb-stage-table.component.tsx

```diff
@@ -226,27 +227,50 @@ export class FtbStageTable {
   }
 
   private renderBody() {
-    let sliceStart = 0;
-    let sliceEnd = this.stage.table.length;
+    let res = [];
 
-    if (this.rowsLimit && this.rowsLimit.limit < this.stage.table.length) {
-      if (!this.rowsLimit.baseTeam || !this.stage.table.some(row => row.team._id == this.rowsLimit.baseTeam._id)) {
+    const getPositions = (baseTeam: Team) => {
+      let sliceStart;
+      let sliceEnd;
+      if (!baseTeam || !this.stage.table.some(row => row.team._id == baseTeam._id)) {
         sliceStart = 0;
         sliceEnd = this.rowsLimit.limit;
         if (this.stage.table.length - 1 === sliceEnd) sliceEnd++; //showing full table if only one row left
       } else {
-        const basePosition = this.stage.table.findIndex(row => row.team._id == this.rowsLimit.baseTeam._id);
+        const basePosition = this.stage.table.findIndex(row => row.team._id == baseTeam._id);
         sliceStart = Math.round(Math.max(0, basePosition - this.rowsLimit.limit / 2));
         sliceEnd = Math.min(sliceStart + this.rowsLimit.limit, this.stage.table.length);
         sliceStart = sliceEnd - this.rowsLimit.limit;
       }
+      return this.stage.table.slice(sliceStart, sliceEnd);
```

Comment:
I think, `getPositions` method should return position array like `[0,1,2]` then we can find intersections with `lodash.intersection` and only then map to table rows